main.go

```diff
@@ -78,102 +246,285 @@ func fatal(formatter string, args ...interface{}) {
-		if err == io.EOF {
-			fatal("couldn't read input event - check stdin must be enabled")
+	check := sensu.NewGoCheck(&plugin.PluginConfig, options, checkArgs, executeCheck, useStdin)
+	check.Execute()
+}
+
+func removeDuplicates(elements []string) []string { // change string to int here if required
+	// Use map to record duplicates as we find them.
+	encountered := map[string]bool{} // change string to int here if required
+	result := []string{}             // change string to int here if required
+
+	for v := range elements {
+		if encountered[elements[v]] {
+			// Do not add duplicate.
+		} else {
+			// Record this element as an encountered element.
+			encountered[elements[v]] = true
+			// Append to result slice.
+			result = append(result, elements[v])
 		}
-		fatal("error decoding input event: %s", err)
 	}
+	// Return the new slice.
+	return result
+}
+
+func buildLogArray() error {
+	var e error
+	if plugin.LogFile != "" {
+		absPath, _ := filepath.Abs(plugin.LogFile)
+		if filepath.IsAbs(absPath) {
+			logs = append(logs, absPath)
+		} else {
+			return fmt.Errorf("Path %s not absolute", absPath)
+		}
 
-	f, err := os.Open(*logFile)
-	if err != nil {
-		fatal("couldn't open log file: %s", err)
 	}
-	defer func() {
-		if err := f.Close(); err != nil {
-			fatal("error closing log file: %s", err)
+	if len(plugin.LogPath) > 0 && len(plugin.LogFileExpr) > 0 {
+		logRegExp, e := regexp.Compile(plugin.LogFileExpr)
+		if e != nil {
+			return e
 		}
-	}()
+		absLogPath, _ := filepath.Abs(plugin.LogPath)
 
-	state, err := getState(*stateFile)
-	if err != nil {
-		fatal("%s", err)
+		if filepath.IsAbs(absLogPath) {
+			e = filepath.Walk(absLogPath, func(path string, info os.FileInfo, err error) error {
+				if err == nil && logRegExp.MatchString(info.Name()) {
+					if filepath.IsAbs(path) {
+						logs = append(logs, path)
+					} else {
+						return fmt.Errorf("Path %s not absolute", path)
+					}
+				}
+				return nil
+			})
+			if e != nil {
+				return e
+			}
+		}
+	}
+	logs = removeDuplicates(logs)
+	if plugin.Verbose {
+		log.Printf("Log file array to process: %v", logs)
 	}
+	return e
+}
+
+func executeCheck(event *types.Event) (int, error) {
+	e := buildLogArray()
+	if e != nil {
+		return sensu.CheckStateCritical, e
+	}
+	file_errors := []string{}
+
+	eventBuf := new(bytes.Buffer)
+	enc := json.NewEncoder(eventBuf)
+
+	status := sensu.CheckStateOK
 
-	// supress alerts on first run (when state file is empty) only when configured (with -ignore-initial-run)
-	if state == (State{}) && *ignoreInitialRun {
+	for _, file := range logs {
+		if !filepath.IsAbs(file) {
+			file_errors = append(file_errors, file)
+			log.Printf("error file %s: is not absolute path", file)
+			continue
+		}
+		if plugin.Verbose {
+			log.Printf("Processing: %v", file)
+		}
+		f, err := os.Open(file)
+		if err != nil {
+			file_errors = append(file_errors, file)
+			log.Printf("error couldn't open log file %s: %s", file, err)
+			continue
```

Comment:
The problem is, this wipes out the match state in the generated events. Because the generated events are by design not resolvable... if you do a missing file critical generated event.. you won't ever automatically resolve that if the file comes back with no matches.  
The place for this sort of error is in the calling event because it doesn't break the auto-resolve pattern.